### PR TITLE
Refactor download script 

### PR DESCRIPTION
**Pull request by ewlu**
- separate download and compare. 
- update aggregate summary info

### CODE REVIEW

**Comment by ewlu** [comment]
testing with https://github.com/patrick-rivos/riscv-gnu-toolchain/actions/runs/5907137349

**Comment by patrick-rivos** [comment]
Are we sure the right branch was tested? #202 does not generate with the updated format:
```Failures Across All Affected Targets ({num_failures} targets / {len(failures)} total targets)```

File: scripts/download_artifacts.py [hunk 1]
```diff
@@ -141,42 +141,13 @@ def download_and_compare_all_artifacts(current_hash: str, token: str):
...
-                f"./logs/{artifact_name.format(base_hash)}",
-                current_hash,
-                f"./logs/{artifact}",
-                compare_path,
-            )
-        except (RuntimeError, ValueError) as err:
-            with open("./logs/failed_testsuite.txt", "a+") as f:
-                f.write(f"{artifact}|{err}\n")
-
+        download_artifact(artifact_name.format(base_hash), str(base_id), token, "previous_logs")
```
Comment (patrick-rivos) [inline scripts/download_artifacts.py:146]: Rather than the if->continue, maybe just do `if base_hash != "No valid hash":`

**Comment by ewlu** [comment]
retesting with https://github.com/patrick-rivos/riscv-gnu-toolchain/actions/runs/5908134177

**Comment by ewlu** [comment]
Testing again on https://github.com/patrick-rivos/riscv-gnu-toolchain/actions/runs/5930340850

**Comment by ewlu** [comment]
last test: https://github.com/patrick-rivos/riscv-gnu-toolchain/actions/runs/5941263306/job/16111808443

**Comment by patrick-rivos** [comment]
LGTM.

**Pull request merged**